### PR TITLE
feat(native/Bottomsheet): add sectionList support

### PR DESCRIPTION
**Pull request by shubhnik**

### CODE REVIEW

File: src/molecules/BottomSheet/BottomSheet.native.js [hunk 1]
```diff
@@ -191,9 +202,10 @@ const BottomSheet = ({
       panGestureComponentEnabled={true}
       adjustToContentHeight={adjustToContentHeight}
       alwaysOpen={initialHeight}
-      rootStyle={styles.rootStyle({ theme })}
+      rootStyle={styles.rootStyle({ theme, isSectionList })}
+      sectionListProps={isSectionList ? contentComponent[0].props.children.props : undefined}
```
Comment (chaitanyadeorukhkar) [inline src/molecules/BottomSheet/BottomSheet.native.js:206]: This is a major assumption that contentComponent's 1'st child will always be a SectionList. Wouldn't it be better for the API to have a `<BottomSheet.SectionList />` and we filter and find it and use that here?

File: src/molecules/BottomSheet/BottomSheetSectionList.native.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { SectionList } from 'react-native';
+
+const BottomSheetSectionList = () => null;
+
+BottomSheetSectionList.displayName = 'BottomSheetSectionList';
+
+/**Not sure if this works :( */
+BottomSheetSectionList.propTypes = SectionList;
```
Comment (chaitanyadeorukhkar) [inline src/molecules/BottomSheet/BottomSheetSectionList.native.js:8]: I dont think it does anything. Not sure what to do here either

Comment (shridhar52war) [inline src/molecules/BottomSheet/BottomSheetSectionList.native.js:8]: This throws waring. Instead, we can define defaultProps?
`BottomSheetSectionList.defaultProps = SectionList;`

File: src/molecules/BottomSheet/__tests__/BottomSheet.native.test.js [hunk 3]
```diff
@@ -241,4 +264,93 @@ describe('<BottomSheet />', () => {
       ),
     ).toThrow('expected onClose prop for `BottomSheet`');
   });
+
+  it('renders BottomSheet with sectionList', () => {
```
Comment (shridhar52war) [inline src/molecules/BottomSheet/__tests__/BottomSheet.native.test.js:268]: Can we add one more story with BottomSheet with `sectionList` and `Footer`?

File: src/molecules/BottomSheet/BottomSheet.native.js [hunk 4]
```diff
@@ -186,14 +204,17 @@ const BottomSheet = ({
...
       onPositionChange={onChange}
       onClosed={handleBottomSheetClose}
-      childrenStyle={styles.childrenStyle({ theme })}
+      childrenStyle={styles.childrenStyle({ theme, isSectionList: Boolean(sectionListComponent) })}
       withHandle={false}
       panGestureComponentEnabled={true}
       adjustToContentHeight={adjustToContentHeight}
       alwaysOpen={initialHeight}
       rootStyle={styles.rootStyle({ theme })}
+      sectionListProps={sectionListComponent ? sectionListComponent[0].props : undefined}
```
Comment (kamleshchandnani) [inline src/molecules/BottomSheet/BottomSheet.native.js:213]:
```suggestion
      sectionListProps={sectionListComponent[0]?.props}
```

File: src/molecules/BottomSheet/__tests__/BottomSheet.native.test.js [hunk 5]
```diff
@@ -241,4 +264,93 @@ describe('<BottomSheet />', () => {
...
+            </View>
+          )}
+          keyExtractor={(item) => item}
+        />
+      </BottomSheet>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should throw error when Content and SectionList are rendered', () => {
```
Comment (kamleshchandnani) [inline src/molecules/BottomSheet/__tests__/BottomSheet.native.test.js:290]:
```suggestion
  it('should throw error when both Content and SectionList are provided', () => {
```

File: src/molecules/BottomSheet/__tests__/BottomSheet.native.test.js [hunk 6]
```diff
@@ -241,4 +264,93 @@ describe('<BottomSheet />', () => {
...
+            </View>
+          </BottomSheet.Content>
+        </BottomSheet>,
+      ),
+    ).toThrow(
+      'expected to have one of `BottomSheet.Content or BottomSheet.SectionList` but found both',
+    );
+  });
+
+  it('should throw error when multiple SectionList are passed', () => {
```
Comment (kamleshchandnani) [inline src/molecules/BottomSheet/__tests__/BottomSheet.native.test.js:320]:
```suggestion
  it('should throw error when multiple SectionList are provided', () => {
```

File: src/molecules/BottomSheet/__tests__/BottomSheet.native.test.js (hunk 3 above)
Comment (kamleshchandnani) [inline src/molecules/BottomSheet/__tests__/BottomSheet.native.test.js:268]:
```suggestion
  it('renders BottomSheet with SectionList', () => {
```

File: src/molecules/BottomSheet/BottomSheetSectionList.native.js (hunk 2 above)
Comment (kamleshchandnani) [reply src/molecules/BottomSheet/BottomSheetSectionList.native.js:8]: why do you need this in first place?

File: src/molecules/BottomSheet/BottomSheetSectionList.native.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { SectionList } from 'react-native';
```
Comment (kamleshchandnani) [inline src/molecules/BottomSheet/BottomSheetSectionList.native.js:1]:
```suggestion
import { SectionList as BottomSheetSectionList } from 'react-native';
BottomSheetSectionList.displayName = 'BottomSheetSectionList';
```

and then you don't need anything else I guess

File: src/molecules/BottomSheet/BottomSheetSectionList.native.js (hunk 2 above)
Comment (shridhar52war) [reply src/molecules/BottomSheet/BottomSheetSectionList.native.js:8]: Yes, right, it does nothing. We can remove this.

File: src/molecules/BottomSheet/BottomSheet.native.js [hunk 8]
```diff
@@ -37,10 +38,14 @@ const styles = {
...
     };
   },
-  childrenStyle: ({ theme }) => {
-    return {
+  childrenStyle: ({ theme, isSectionList }) => {
+    const childrenStyles = {
       backgroundColor: theme.colors.background[200],
     };
+    if (isSectionList) {
+      return { ...childrenStyles, ...{ paddingVertical: 8, paddingHorizontal: 16 } };
```
[the remark this answers is not in the document]
Comment (shubhnik) [reply src/molecules/BottomSheet/BottomSheet.native.js:46]: This is the padding for `<BottomSheet.Content/>`, will be required here too...

File: src/molecules/BottomSheet/BottomSheetSectionList.native.js (hunk 2 above)
Comment (shubhnik) [reply src/molecules/BottomSheet/BottomSheetSectionList.native.js:8]: Needed it to pass RN SectionList propTypes to our BottomSheetSectionList.

File: src/molecules/BottomSheet/BottomSheet.native.js [hunk 9]
```diff
@@ -85,6 +90,7 @@ const BottomSheet = ({
   const headerComponent = bottomsheetChildrenGroupByDisplayName.BottomSheetHeader;
   const footerComponent = bottomsheetChildrenGroupByDisplayName.BottomSheetFooter;
   const contentComponent = bottomsheetChildrenGroupByDisplayName.BottomSheetContent;
+  const sectionListComponent = bottomsheetChildrenGroupByDisplayName.BottomSheetSectionList;
```
[the remark this answers is not in the document]
Comment (shubhnik) [reply src/molecules/BottomSheet/BottomSheet.native.js:93]: `sectionLisComponent` could be undefined but the check isn't needed here. We check it where it is being used further.

[the remark this answers is not in the document]
Comment (shubhnik) [reply src/molecules/BottomSheet/BottomSheet.native.js:93]: `sectionListComponent` could be undefined but the check isn't needed here.

**Review by chaitanyadeorukhkar — APPROVED** [review]

**Review by shridhar52war — APPROVED** [review]

**Pull request closed without merging**